tf/edgeml/graph/protoNN.py

```diff
@@ -0,0 +1,141 @@
+'''
+ProtoNN Graph construction
+'''
+import numpy as np
+import tensorflow as tf
+
+
+class ProtoNN:
+    def __init__(self, inputDimension, projectionDimension, numPrototypes,
+                 numOutputLabels, gamma,
+                 W = None, B = None, Z = None):
+        '''
+        inputDimension: input data dimension. A [-1, inputDimension] matrix
+            is expected as part of __call__ method.
+        projectionDimension: (model paramter)
+        numPrototypes: (model parameter)
+        numOutputLabels: The number of output labels or classes
+        W, B, Z: initial value to use for W, B, Z
+        Expected Dimensions:
+        W   d x d_cap
+        B   d_cap x m
+        Z   L x m
+        X   n x d
+        '''
+        with tf.name_scope('protoNN') as ns:
+            self.__nscope = ns
+        self.__d = inputDimension
+        self.__d_cap = projectionDimension
+        self.__m = numPrototypes
+        self.__L = numOutputLabels
+
```

Comment:
Change the variable names to more readable ones like inputDims, projDims etc.,
try not to use single letters as above

Reply:
Variables with `__` are internal and private. I'll change it anyway.